Northstar.CustomServers/mod/scripts/vscripts/evac/_evac.gnut

```diff
@@ -428,3 +494,89 @@ void function EvacDropshipKilled( entity dropship, var damageInfo )
 		}
 	}
 }
+
+// if there's no player left in evacing team, we end this match
+void function CheckIfAnyPlayerLeft( int evacTeam )
+{
+	wait GAME_EPILOGUE_PLAYER_RESPAWN_LEEWAY
+	float startTime = Time()
+
+	OnThreadEnd(
+		function() : ( evacTeam )
+		{
+			SetTeamActiveObjective( evacTeam, "EG_DropshipExtractEvacPlayersKilled" )
+			SetTeamActiveObjective( GetOtherTeam( evacTeam ), "EG_StopExtractEvacPlayersKilled" )
+			thread EvacEpilogueCompleted( null )
+		}
+	)
+	while( true )
+	{
+		if( GetPlayerArrayOfTeam_Alive( evacTeam ).len() == 0 )
+			break
+		if( GetGameState() == eGameState.Postmatch )
+			return
+		WaitFrame()
+	}	
+}
+
+void function DropShipTempHide( entity dropship )
+{
+	dropship.kv.VisibilityFlags = 0 // or it will still shows the jetpack fxs
+	HideName( dropship )
+	wait 0.46
+	if( IsValid( dropship ) )
+	{
+		dropship.kv.VisibilityFlags = ENTITY_VISIBLE_TO_EVERYONE
+		ShowName( dropship )
+	}
+}
+
+EvacShipSetting function GetEvacShipSettingByTeam( int team )
+{
+	EvacShipSetting tempSetting
+	if( team == TEAM_IMC )
+	{
+		tempSetting.shipModel = $"models/vehicle/goblin_dropship/goblin_dropship_hero.mdl"
+		tempSetting.flyinSound = "Goblin_IMC_Evac_Flyin"
+		tempSetting.hoverSound = "Goblin_IMC_Evac_Hover"
+		tempSetting.flyoutSound = "Goblin_IMC_Evac_FlyOut"
+	}
+	if( team == TEAM_MILITIA )
+	{
+		tempSetting.shipModel = $"models/vehicle/crow_dropship/crow_dropship_hero.mdl"
+		tempSetting.flyinSound = "Crow_MCOR_Evac_Flyin"
+		tempSetting.hoverSound = "Crow_MCOR_Evac_Hover"
+		tempSetting.flyoutSound = "Crow_MCOR_Evac_Flyout"
+	}
+	return tempSetting
+}
+
+void function EvacDropshipDamaged( entity evacShip, var damageInfo )
+{
+	int damageSourceID = DamageInfo_GetDamageSourceIdentifier( damageInfo )
+	if( damageSourceID == damagedef_nuclear_core )
+		DamageInfo_SetDamage( damageInfo, DamageInfo_GetDamage( damageInfo ) * EVAC_SHIP_DAMAGE_MULTIPLIER_AGAINST_NUCLEAR_CORE )
+}
+
+void function WarpInEffectEvacShip( entity dropship )
+{
+	float sfxWait = 0.1
+	float totalTime = WARPINFXTIME
+	float preWaitTime = 0.16 // give it some time so it's actually playing anim, and we can get it's "origin" attatch for playing warp in effect
+	string sfx = "dropship_warpin"
+
+	wait preWaitTime
+
+	int attach = dropship.LookupAttachment( "origin" )
```

Comment:
Since we are passing an entity and then waiting before doing stuff, there is a chance that somehow the dropship is no longer valid, probably best to add an EndSignal to this function to try and prevent crashes